File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        l=r=maxLen= 0
        mySet = set()
        while r < len(s): 
            while l < r and s[r] in mySet :
                mySet.remove(s[l]) 
                l +=1    
            mySet.add(s[r])
            r +=1
            maxLen = max(maxLen, r-l)

        return maxLen
            
                
        #  set , l ,r
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        int l = 0 , maxLen = 0 ;
        set<char> mySet;
        for(int r =0 ; r < n ; r++){
            while(mySet.count(s[r])){
                mySet.erase(s[l]);
                l++;
            }
            mySet.insert(s[r]);
            maxLen = max(maxLen , r-l+1);
        }

        return maxLen;

    }
};